### PR TITLE
Expand Cluster.Hosting options

### PR DESCRIPTION
**Pull request by Arkatufus**
Part of #142

## Changes
- Expand Cluster.Hosting options
- Add, expand, and clean up XML-DOCs

### CODE REVIEW

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:52]: Don't like the `Role` type - adds more friction to the configuration surface in order to expose a feature barely anybody uses. Users can configure this in HOCON if they really want it - please remove it from the UI surface area entirely and go back to using simple strings.

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:57]: So here's a goal worth considering with the `ClusterOptions` class - should this be parseable from an `appsettings.json` file / Environment Variable? If so, maybe we should support `string` here instead and just handle the parsing on the back-end. I'm leaning towards doing that for 1.0.

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:157]: LGTM

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:162]: All of the `ShardOptions` changes LGTM

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Aaronontheweb) [reply src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:57]: Yeah, let's change this to a `string` type

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Need to lean more towards ease of use / low friction even if it means not exposing certain settings in Akka.Hosting. Goal of this library is low friction, not completeness.

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Aaronontheweb) [reply src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:57]: Here's what we have to do in the current stable version of Akka.Hosting:

```csharp
var akkaSection = context.Configuration.GetSection("Akka");

// maps to environment variable Akka__ClusterIp
var hostName = akkaSection.GetValue<string>("ClusterIp", "localhost");

// maps to environment variable Akka__ClusterPort
var port = akkaSection.GetValue<int>("ClusterPort", 0);

var seeds = akkaSection.GetValue<string[]>("ClusterSeeds", new[] { "akka.tcp://SqlSharding@localhost:7918" })
    .Select(Address.Parse)
    .ToArray();
```

It would be amazing if we could just do this instead (and if we decide to go down this route, we should write a unit test that looks like this):

**`appsettings.json`**

```json
{
    "Logging": {
        "LogLevel": {
            "Default": "Debug",
            "System": "Information",
            "Microsoft": "Information"
        }
    },
    "Akka": {
        "RemoteOptions":{
            "Hostname": "localhost",
            "Port": 8081,
            "PublicHostname": "localhost",
            "PublicPort": 8081
        },
        "ClusterOptions": {
            "SeedNodes": "akka.tcp://ClusterSystem@localhost:8081",
            "Roles": ["backend"]
        }
    }
}
```

**`program.cs`**

```csharp
var akkaSection = context.Configuration.GetSection("Akka");
var clusterConfig = akkaSection.Get<ClusterOptions>(); // ideally, this should work
```

We've not really stated this as an explicit goal for the project before, but in the context of our "pit of success" mission for this library I think it's probably worth doing before we ship 1.0. What do you think?

File: src/Akka.Cluster.Hosting.Tests/ClusterOptionsSpec.cs
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting.Tests/ClusterOptionsSpec.cs:125]: Binding directly from json provided configuration is totally doable now.

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:49]: Would this be a better compromise to the `Role` class?
We're not forcing the user to provide any values and still keeps the class as simple as possible.

**Review by Arkatufus — COMMENTED** [review]
Self review for the last change request

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Arkatufus) [reply src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:49]: ah, wrong line...

File: src/Akka.Cluster.Hosting.Tests/ClusterOptionsSpec.cs
Comment (Aaronontheweb) [reply src/Akka.Cluster.Hosting.Tests/ClusterOptionsSpec.cs:125]: Looks great

File: src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs
Comment (Arkatufus) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:33]: This is the line I ment.
Would this be a better compromise to the Role class?
We're not forcing the user to provide any values and still keeps the class as simple as possible.

Comment (Aaronontheweb) [inline src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs:33]: Yep, I think that's fine

**Review by Aaronontheweb — APPROVED** [review]
LGTM - let's auto-merge this once you resolve your merge conflict

**Pull request merged**
